Userland/Libraries/LibGfx/PortableImageLoaderCommon.h

```diff
@@ -197,76 +184,35 @@ static void set_pixels(TContext& context, Vector<Gfx::Color> const& color_data)
 }
 
 template<typename TContext>
-static bool decode(TContext& context)
+static ErrorOr<void> decode(TContext& context)
 {
     if (context.state >= TContext::State::Decoded)
-        return true;
-
-    auto error_guard = ArmedScopeGuard([&] {
-        context.state = TContext::State::Error;
-    });
+        return {};
 
-    Streamer streamer(context.data, context.data_size);
+    TRY(read_magic_number(context));
 
-    if (!read_magic_number(context, streamer))
-        return false;
+    TRY(read_whitespace(context));
 
-    if (!read_whitespace(context, streamer))
-        return false;
-
-    if (!read_width(context, streamer))
-        return false;
-
-    if (!read_whitespace(context, streamer))
-        return false;
-
-    if (!read_height(context, streamer))
-        return false;
+    TRY(read_width(context));
+    TRY(read_whitespace(context));
+    TRY(read_height(context));
 
     if (context.width > maximum_width_for_decoded_images || context.height > maximum_height_for_decoded_images) {
         dbgln("This portable network image is too large for comfort: {}x{}", context.width, context.height);
-        return false;
+        return Error::from_string_literal("This portable network image is too large for comfort.");
```

Comment:
“for comfort” is really dubious here—it could better without that phrase (one could optionally tell what is the maximum, at least in the debug message?)

Reply:
We shouldn't try to guess if the image is too big or not, and let the kernel decide if we are OOM. But that's outside the scope of this PR.